app/application.jsx

```diff
@@ -4,17 +4,31 @@ import React from 'react';
 import { render } from 'react-dom';
 import { Router, Route, Redirect, browserHistory } from 'react-router';
 import AppLayout from 'components/application/layout';
-import TodoLayout from 'components/todo/layout';
+import Main from 'components/main/view';
 import AboutLayout from 'components/about/layout';
 import About from 'components/about/view';
 import Detailed from 'components/about/detailed';
+import session from 'services/session';
 import globalStyles from 'stylesheets/main';
 
+const requireAuth = (nextState, replace) => {
+  if (!session.loggedIn()) {
+    replace({
+      pathname: '/',
+      state: { nextPathname: nextState.location.pathname }
+    });
+  }
+};
```

Comment:
what do you think about extracting `requireAuth` into separate route helper file and keep here only route definitions?
btw, is it any difference between defining `requireAuth` as `requireAuth = function()` and `function requireAuth()`, just wondering why we have to assign the function definition to const here instead of explicit function definition ?